src/nnet3/nnet-normalize-component.cc

```diff
@@ -579,11 +699,42 @@ void BatchNormComponent::StoreStats(
   if (stats_sum_.Dim() != block_dim_) {
     stats_sum_.Resize(block_dim_);
     stats_sumsq_.Resize(block_dim_);
+    moving_mean_.Resize(block_dim_);
+    moving_stddv_.Resize(block_dim_);
     KALDI_ASSERT(count_ == 0);
   }
   count_ += num_frames;
   stats_sum_.AddVec(num_frames, mean, 1.0);
   stats_sumsq_.AddVec(num_frames, uvar, 1.0);
+
+  // update moving averages
+  if (batch_renorm_) {
+    if (training_begining_) {
+      KALDI_WARN << "Training Begining : " << training_begining_;
+      training_begining_ = false;
+      moving_mean_.CopyFromVec(mean);
+      CuSubVector<BaseFloat> scale(memo->mean_uvar_scale, 2);
+      scale.ApplyPow(-1.0);
+      moving_stddv_.CopyFromVec(scale);
+      scale.ApplyPow(-1.0);
+    } else {
+      CuVector<BaseFloat> mv_mean(moving_mean_), mv_stddv(moving_stddv_); 
+
+      BaseFloat renorm_momentum_2 = 1.0 - renorm_momentum_;
+
+      moving_mean_.Scale(renorm_momentum_2);
+      moving_mean_.AddVec(renorm_momentum_, mean);
+
+      CuSubVector<BaseFloat> scale(memo->mean_uvar_scale, 2);
+      scale.ApplyPow(-1.0);
+
+      moving_stddv_.Scale(renorm_momentum_2);
+      moving_stddv_.AddVec(renorm_momentum_, scale);
+      mv_mean.CopyFromVec(moving_mean_);
```

Comment:
I think this is backwards.  mv_mean is the temporary, why are you copying from the class member?  Or maybe this is old code that's not needed now.